src/Libraries/CoreNodeModels/Selection.cs

```diff
@@ -115,6 +141,28 @@ public virtual string SelectionSuggestion
             ShouldDisplayPreviewCore = true;
         }
 
+        [JsonConstructor]
+        protected SelectionBase(
+            SelectionType selectionType,
+            SelectionObjectType selectionObjectType,
+            string message,
+            string prefix,
+            IEnumerable<string> selectionIdentifier,
+            IEnumerable<PortModel> inPorts,
+            IEnumerable<PortModel> outPorts) : base(inPorts, outPorts)
+        {
```

Comment:
I am also still getting double output ports when deserializing even after adding these parameters.  I know this was due to recent changes in the node model base class but not sure why it is still occurring.  Maybe an empty constructor should created at the `DSFaceSelection` level

Reply:
yeah try that and make sure NodeModel(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts) is called

Reply:
@QilongTang @mjkkirschner  I have actually made some progress by adding JsonConstructors through the entire inheritance chain shown in the description making sure to pass all the original parameters plus the revit element id propety data and port data.  I am now getting a single output and the Revit referenced element is deserializing properly.  The issue now is I can't select a new element after opening the file but I have a few ideas on resolving this.

![image](https://user-images.githubusercontent.com/13341935/29781690-0778a1f4-8be8-11e7-9c11-275aa0980b09.png)

Reply:
@alfarok I believe the comment above is no longer valid? Do you have an update pic of the same DYN?

Reply:
@QilongTang yup this has all been resolved and I verified model elements outside of the standard geometry types are properly restoring as well (wall, roof, etc). 
![giphy](https://user-images.githubusercontent.com/13341935/29971937-3020ffc6-8ef8-11e7-90e2-87ad703034c1.gif)